.github/workflows/CI.yml

```
@@ -493,50 +451,68 @@ jobs:
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          password: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+        env:
+          AWS_REGION: us-east-1
+
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
           aws-region: us-west-2
 
-      - name: restore cached rpms
-        uses: actions/cache@v2
-        with:
-          key: "cached_packages_${{ github.run_id }}"
-          path: build/packages
-
       - name: Create SSM package
         run: |
-          ssm_pkg_version=`cat build/packages/VERSION`
+          ssm_pkg_version=`cat $PACKAGING_ROOT/VERSION`
           (aws ssm describe-document --name "testAWSDistroOTel-Collector" --version-name ${ssm_pkg_version} >/dev/null 2>&1) || {
             pip3 install boto3
             python3 tools/ssm/ssm_manifest.py ${ssm_pkg_version}
-            aws s3 cp build/packages/ssm s3://aws-otel-collector-test/ssm/${ssm_pkg_version} --recursive
+            aws s3 cp $PACKAGING_ROOT/ssm s3://aws-otel-collector-test/ssm/${ssm_pkg_version} --recursive
             python3 tools/ssm/ssm_create.py --no-default testAWSDistroOTel-Collector ${ssm_pkg_version} aws-otel-collector-test/ssm/${ssm_pkg_version} us-west-2
           }
 
       - name: Upload to S3 in the testing stack
         if: steps.e2etest-release.outputs.cache-hit != 'true'
         run: s3_bucket_name=aws-otel-collector-test upload_to_latest=0 bash tools/release/image-binary-release/s3-release.sh
 
-      - name: Login to Public ECR
+      #Build the adot collector image for two primary reasons:
+      #-Using the adot collector image to do the integration test
+      #-Export it for delivery version image in CD
+      #Documentation: https://github.com/docker/build-push-action
+      - name: Build adot collector image
+        uses: docker/build-push-action@v2
         if: steps.e2etest-release.outputs.cache-hit != 'true'
-        id: login-ecr
-        uses: docker/login-action@v1
         with:
-          registry: public.ecr.aws
-          username: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
-          password: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
-        env:
-          AWS_REGION: us-east-1
-
-      - name: upload to ECR
-        if: steps.e2etest-release.outputs.cache-hit != 'true'
-        run: |
-          docker load < build/packages/$IMAGE_NAME.tar
-          docker tag $IMAGE_NAME public.ecr.aws/$ECR_REPO:${{ needs.e2etest-preparation.outputs.version }}
-          docker push public.ecr.aws/$ECR_REPO:${{ needs.e2etest-preparation.outputs.version }}
+          file: cmd/awscollector/Dockerfile
+          context: .
+          push: true
+          tags: public.ecr.aws/${{ env.ECR_REPO }}:${{ needs.e2etest-preparation.outputs.version }}
```

```suggestion
          tags: public.ecr.aws/$ECR_REPO:${{ needs.e2etest-preparation.outputs.version }}
```
nitpick: For consistency.

---

Cannot used this since I tested it once and [get an error with it though](https://github.com/khanhntd/aws-otel-collector/runs/4734505542?check_suite_focus=true). (an example is a similar-one but yeah its works the same)

---

That's interesting. How does it work in the CD then?

---

[Not worked and having a similar error as expected though](https://github.com/khanhntd/aws-otel-collector/runs/4767860865?check_suite_focus=true). Fixed it with the newest commit